HelpMyStreetFE/HelpMyStreetFE/Controllers/CommunityController.cs

```diff
@@ -43,7 +43,29 @@ public CommunityController(ILogger<CommunityController> logger, ICommunityReposi
             _groupMemberService = groupMemberService;
         }
 
-        public async Task<IActionResult> Index(string groupKey, CancellationToken cancellationToken)
+        [AuthorizeAttributeNoRedirect]
+        [Route("community/joinandgo/{groupIdEnc}")]
+        public async Task<IActionResult> JoinAndGo(string groupIdEnc, CancellationToken cancellationToken)
+        {
+            var groupId = Int32.Parse(Base64Utils.Base64Decode(groupIdEnc));
+            var group = (Groups)groupId;
+
+            CommunityViewModel communityViewModel = await _communityRepository.GetCommunity(groupId, cancellationToken);
+
+            if (communityViewModel == null)
+            {
+                return RedirectToAction("Error404", "Errors");
+            }
+
+            var user = await _authService.GetCurrentUser(HttpContext, cancellationToken);
+            var outcome = await _groupMemberService.PostAssignRole(user.ID, groupId, GroupRoles.Member, -1, cancellationToken);
```

Comment:
I believe this role will already be assigned by AddUserToDefaultGroups, so we may not need the new route...